pkg/endpoint/policy.go

```diff
@@ -18,86 +18,50 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"reflect"
 
 	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/pkg/labels"
 	"github.com/cilium/cilium/pkg/logfields"
 	"github.com/cilium/cilium/pkg/policy"
 	"github.com/cilium/cilium/pkg/policy/api"
-	"github.com/cilium/cilium/pkg/u8proto"
 
 	"github.com/cilium/cilium/common"
 	log "github.com/sirupsen/logrus"
 )
 
-func (e *Endpoint) checkEgressAccess(owner Owner, opts models.ConfigurationMap, dstID policy.NumericIdentity, opt string) {
-	var err error
-
+func (e *Endpoint) checkEgressAccess(owner Owner, dstLabels labels.LabelArray, opts models.ConfigurationMap, opt string) {
 	ctx := policy.SearchContext{
 		From: e.Consumable.LabelArray,
+		To:   dstLabels,
```

Comment:
I was a bit surprised that this wasn't called out specifically in the patch description or indeed a separate patch, so it seems like this was a bug in that previously egress label-based access couldn't be properly applied with to:host or to:world restrictions. Are there tests for this?

Reply:
ctx.To was assigned separately, see some of the deleted lines below. There should be no functional change here.